### PR TITLE
enforce allowlist policy on transfers

### PR DESCRIPTION
**Pull request by ericfranz**
this was accidentally not committed; could use tests (along with the others) but maybe a separate issue for that

### CODE REVIEW

**Review by johrstrom — APPROVED** [review]
:+1:

File: apps/dashboard/app/models/transfer.rb [hunk 1]
```diff
@@ -7,11 +7,16 @@ class Transfer
   attr_accessor :id, :status, :created_at, :started_at, :completed_at
   attr_writer :percent
 
-  validates_each :files do |record, attr, value|
+  validates_each :files do |record, attr, files|
     if record.action == 'mv' || record.action == 'cp'
-      conflicts = value.values.select {|f| File.exist?(f) }
+      conflicts = files.values.select {|f| File.exist?(f) }
       record.errors.add :files, "these files already exist: #{conflicts.join(', ')}" if conflicts.present?
```
Comment (johrstrom) [inline apps/dashboard/app/models/transfer.rb:13]: This is not really a part of this change, I'm just noticing it now. We're not allowing moves or copies if the destination already exists? `cp` and `mv` allow for this just fine, is there a reason we're differing from their behaviour?

Comment (ericfranz) [reply apps/dashboard/app/models/transfer.rb:13]: Preexisting functionality. We can remove this limitation if we enhance the user interface to provide warning to the user. Typically if a move or copy would end up overwriting existing files and you wanted to do that you would first be presented with a warning and an option to proceed. That will just take more time to get right.

Comment (ericfranz) [reply apps/dashboard/app/models/transfer.rb:13]: Note I think you could open an issue about adding supporting for that.

**Pull request merged**